homeassistant/components/hydrawise/sensor.py

```diff
@@ -44,21 +47,20 @@ def update(self):
         """Get the latest data and updates the states."""
         mydata = self.hass.data[DATA_HYDRAWISE].data
         _LOGGER.debug("Updating Hydrawise sensor: %s", self._name)
+        relay_data = mydata.relays[self.data["relay"] - 1]
         if self._sensor_type == "watering_time":
-            if not mydata.running:
-                self._state = 0
+            if relay_data["timestr"] == "Now":
+                self._state = int(relay_data["run"] / 60)
             else:
-                if int(mydata.running[0]["relay"]) == self.data["relay"]:
-                    self._state = int(mydata.running[0]["time_left"] / 60)
-                else:
-                    self._state = 0
+                self._state = 0
         else:  # _sensor_type == 'next_cycle'
-            for relay in mydata.relays:
-                if relay["relay"] == self.data["relay"]:
-                    if relay["nicetime"] == "Not scheduled":
-                        self._state = "not_scheduled"
-                    else:
-                        self._state = f"{relay['nicetime'].split(',')[0]} {relay['nicetime'].split(' ')[3]}"
+            if relay_data["time"] > ONE_YEAR_SECONDS:
+                self._state = "not_scheduled"
+            else:
+                _LOGGER.debug("New cycle time: %s", relay_data["time"])
+                self._state = time.asctime(
```

Comment:
If we're making a breaking change anyway we can clean up this sensor and make it use the `timestamp` device class. Isn't that what this state is, an absolute time stamp? We just need to format the time correctly and make it UTC time.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
OK. I added the ```timestamp``` device class for the sensor. While I was at it I added the ```switch``` device class to the switches too. I also moved the ```device_class()``` and ```icon()``` methods to the ```HydrawiseEntity()``` class to remove duplicates and clean the code up a bit.

Is there other changes you were asking for in regards to formatting the time for UTC? If so do you have a suggestion for what I need to do?

Reply:
With that change I'm getting an error, ```OverflowError: timestamp out of range for platform time_t```. I only see it on my raspberry pi rig, and not my laptop (Mac) dev environment. Let me investigate that before going further. 

The sensor will give a timestamp that's 50 years away in some cases. I think that's causing an overflow on 32-bit systems like the raspberry pi? Not sure.

Reply:
Capping timestamps to be 2 years or less into the future stops the overflow errors. This feels hacky, but works.